common.sh

```diff
@@ -187,9 +187,35 @@ create_instance()
 get_instance_ip()
 {
     execution_seq=$((${execution_seq}+1))
-    INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
-                        --query "Reservations[*].Instances[*].PrivateIpAddress" \
-                        --output=text)
+    local retry=20
+    local ret=0
+    local bash_option=$-
+    local restore_e=0
+    if [[ $bash_option =~ e ]]; then
+        restore_e=1
+        set +e
+    fi
+    while [ $retry -ge 0 ]; do
+        INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
+                            --query "Reservations[*].Instances[*].PrivateIpAddress" \
+                            --output=text)
+        ret=$?
+        if [ "$ret" -eq 0 ] && [ -n "$INSTANCE_IPS" ]; then
+            echo "Successfully get instance ips: ${INSTANCE_IPS}."
+            break
+        else
+            echo "Failed to get instance ips, retry after 10 seconds."
```

Comment:
Is this really a helpful debugging print?

Reply:
Fixed in latest revision. I remove the printing in the loop but put it to the end